happi/tests/test_prompt.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+
+from happi.prompt import enforce_list, read_user_dict
+
+
+def test_user_dict(runner):
+    default_dict = {'default_key': 'default_value'}
+
+    # normal operation
+    with runner.isolation('key1\nvalue1'):
+        result = read_user_dict('prompt', default=default_dict)
+
+    assert result == {'key1': 'value1'}
+
+    # read default
+    with runner.isolation('\n'):
+        result = read_user_dict('prompt', default=default_dict)
+
+    assert result == default_dict
+
+    # reject keywords
+    with runner.isolation('is\nnotis\n1\n'):
+        result = read_user_dict('prompt', default=default_dict)
+
+    assert result == {'notis': 1}
+
+    # replace values
+    with runner.isolation('key\n1\nkey\n2'):
+        result = read_user_dict('prompt', default=default_dict)
+
+    assert result == {'key': 2}
+
+
+@pytest.mark.parametrize('user_in', (
+    ['a', 'b', 2, 3],
+    "['a', 'b', 2, 3]"
+))
```

Comment:
Would be good practice to get some failing test cases in here - mainly for test coverage.